frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, Renderer2, inject } from '@angular/core';
+import { NgxOtpInputConfig } from "ngx-otp-input";
+import { EndpointService } from '../../services/endpoint/endpoint.service';
+import { ApiBaseService } from '../../services/base-api/api-base.service';
+import { Router } from '@angular/router';
+import { Location } from '@angular/common';
+import { StateService } from '../../services/state/state.service';
+import { Subscription, interval } from 'rxjs';
+
+@Component({
+  selector: 'lib-otp',
+  templateUrl: './otp.component.html',
+  styleUrl: './otp.component.css'
+})
+export class OtpComponent {
+  configData: any;
+  endPointService: EndpointService;
+  location: Location;
+  otpInput: boolean = false;
+  regFormData: any;
+  otp!: any;
+  checkbox: boolean = false;
+
+  baseApiService: ApiBaseService;
+  router: Router;
+  timeLeft: number = 60;
+
+  constructor(private stateService: StateService, private renderer:Renderer2) {
+    this.baseApiService = inject(ApiBaseService);
+    this.endPointService = inject(EndpointService);
+    this.router = inject(Router);
+    this.location = inject(Location);
+  }
+
+  ngOnInit() {
+    this.startTimer();
+    this.regFormData = this.stateService.getData();
+    this.fetchConfigData();
+  }
+
+  async fetchConfigData() {
+    try {
+      this.configData = await this.endPointService.getEndpoint();
+    } catch (error) {
+      console.error("An error occurred while fetching configData:", error);
+    }
+  }
+
+  otpInputConfig: NgxOtpInputConfig = {
+    otpLength: 6,
+    autofocus: true
+  };
+
+
+  handeOtpChange(value: string[]): void {
+    const otp = value.join('');
+    if (otp.length !== 6) {
+      this.otp = "";
+    }
+  }
+
+  handleFillEvent(value: string): void {
+    this.otp = value;
+  }
+
+  navigateBack() {
+    this.location.back();
+  }
+
+  processOTP(action: 'generate' | 'verify') {
```

Comment:
Good however this logic can still needs to be refined as there' scope for generalised due to overlapping logic